lib/mrtrix3/dwi2mask/synthstrip.py

```diff
@@ -0,0 +1,114 @@
+  options.add_argument('-mo',action='store_true',default=False,help='Alternative model weights')
+  options.add_argument('-nocsf',action='store_true', default=False, help='Compute the immediate boundary of brain matter excluding surrounding CSF')
+  options.add_argument('-b', type=int,help='Control the boundary distance from the brain')
+  parser.add_argument('-clean_scale',
+                      type=int,
+                      default=DEFAULT_CLEAN_SCALE,
+                      help='the maximum scale used to cut bridges. A certain maximum scale cuts '
+                           'bridges up to a width (in voxels) of 2x the provided scale. Setting '
+                           'this to 0 disables the mask cleaning step. (Default: ' + str(DEFAULT_CLEAN_SCALE) + ')')
+  
+
+
+def get_inputs(): #pylint: disable=unused-variable
+  pass
+
+
+
+def needs_mean_bzero(): #pylint: disable=unused-variable
+  return True
+
+
+
+def execute(): #pylint: disable=unused-variable
+  
+  run.command('dwiextract input.mif - -bzero | mrmath - mean mean_bzero.mif -axis 3')
+  
+  run.command('mrconvert mean_bzero.mif 3dbzero.mif -axes 0,1,2')
+  app.cleanup('mean_bzero.mif')
+  run.command('mrconvert 3dbzero.mif 3dbzero.nii')
+  app.cleanup('3dbzero.mif')
+  
+  
+  synthstrip_cmd = shutil.which("mri_synthstrip")
+  if not synthstrip_cmd:
+    raise MRtrixError('Unable to locate "Synthstrip" executable; please check installation')
+  
+  cmd_string =SYNTHSTRIP + ' -i' +' 3dbzero.nii' + ' -m' +' synthstrip_mask.nii'
+  output_file='synthstrip_mask.mif'
+  #current_path=os.path.abspath('input.mif')
+  current_path=os.path.abspath(__file__)
+  father_path=os.path.abspath(os.path.dirname(current_path))
+
+
+  if app.ARGS.h:
+    cmd_string=SYNTHSTRIP +' -h'
+    warnings.warn('Displaying help message will not produce any desired files,the output of this command only produce the original input file with desired output file name, if need files, please rerun the command without -h syntax')
+  if app.ARGS.s:
+    cmd_string+=' -o '+ father_path+'/stripped.nii'
+    warnings.warn('The stripped file will be saved in mrtrix3/lib/mrtrix3/dwi2mask')
+  if app.ARGS.g:
+    cmd_string+=' -g'
+    
+  if app.ARGS.nocsf:
+    cmd_string+=' --no-csf'
+
+  if app.ARGS.b is not None:
+    cmd_string += ' -b' + ' '+str(app.ARGS.b)
+
+  if app.ARGS.mo:
+    cmd_string+= ' --model' + father_path + ' -model.nii'
+
+  #try:
+  #  #run.command( 'mri_synthstip -i 3dbzero.nii -o' + str(app.ARGS.o)+ '-m' + str(app.ARGS.m)+'-no-csf'+str(app.ARGS.no-csf)+'-b'+str(app.ARGS.b))
+  #  run.command( 'mri_synthstrip -i 3dbzero.nii -o stripped.nii -m synthstrip_mask.nii')
+  #except run.MRtrixError:
+  #  app.warn('The input image may have more than 1 frames.')
+  if app.ARGS.h:
+    run.command(cmd_string)
+    return 'input.mif'
+  else:
+    run.command(cmd_string)
+    run.command('mrconvert synthstrip_mask.nii synthstrip_mask.mif -datatype bit')
```

Comment:
If variable `output_file` is created to store path "`synthstrip_mask.mif`", the code should make use of it. Otherwise, you could erroneously modify one but not the other, and the script would stop working; whereas if you use the path as stored in the variable throughout, you're protected against introducing that sort of problem.